Sources/SwiftInspectorCore/Tests/PropertyAnalzyerSpec.swift

```diff
@@ -0,0 +1,468 @@
+        }
+      }
+
+      context("when there is a line comment") {
+        beforeEach {
+          let content = """
+          public final class FakeType {
+            // The thing
+            public var thing: String = "Hello, World"
+          }
+          """
+          fileURL = try? Temporary.makeFile(content: content)
+        }
+
+        it("detects the property with the comment") {
+          let result = try? sut.analyze(fileURL: fileURL)
+          expect(result?.properties) == [
+            TypeWithPropInfo.PropInfo(
+              name: "thing",
+              typeAnnotation: "String",
+              comment: "// The thing",
+              access: .public,
+              scope: .instance)
+          ]
+        }
+      }
+
+      context("when there is a doc line comment") {
+        beforeEach {
+          let content = """
+          public final class FakeType {
+            /// The thing
+            public var thing: String = "Hello, World"
+          }
+          """
+          fileURL = try? Temporary.makeFile(content: content)
+        }
+
+        it("detects the property with the comment") {
+          let result = try? sut.analyze(fileURL: fileURL)
+          expect(result?.properties) == [
+            TypeWithPropInfo.PropInfo(
+              name: "thing",
+              typeAnnotation: "String",
+              comment: "/// The thing",
+              access: .public,
+              scope: .instance)
+          ]
+        }
+      }
+
+      context("when there is a block comment") {
+        beforeEach {
+          let content = """
+          public final class FakeType {
+            /* The thing */
+            public var thing: String = "Hello, World"
+          }
+          """
+          fileURL = try? Temporary.makeFile(content: content)
+        }
+
+        it("detects the property with the comment") {
+          let result = try? sut.analyze(fileURL: fileURL)
+          expect(result?.properties) == [
+            TypeWithPropInfo.PropInfo(
+              name: "thing",
+              typeAnnotation: "String",
+              comment: "/* The thing */",
+              access: .public,
+              scope: .instance)
+          ]
+        }
+      }
+
+      context("when there is a doc block comment") {
+        beforeEach {
+          let content = """
+          public final class FakeType {
+            /** The thing */
+            public var thing: String = "Hello, World"
+          }
+          """
+          fileURL = try? Temporary.makeFile(content: content)
+        }
+
+        it("detects the property with the comment") {
+          let result = try? sut.analyze(fileURL: fileURL)
+          expect(result?.properties) == [
+            TypeWithPropInfo.PropInfo(
+              name: "thing",
+              typeAnnotation: "String",
+              comment: "/** The thing */",
+              access: .public,
+              scope: .instance)
+          ]
+        }
+      }
+
+    }
+  }
+}
```

Comment:
It could be nice to test that this works correctly when a property has an attribute.

```swift
class Foo {
  @objc let hello = "world"
}
```

Reply:
And we may even want to include the attribute in our result.

Reply:
hmm would it be ok to add that in a follow up? This feels like a use case that would be uncommon to need. I will add a test for it, but I don't want to add it to the output at this point.

Reply:
I agree, let's just add it as a follow-up if needed.

Reply:
That's fine by me.